### PR TITLE
Share Extension dashboard notification logic

### PR DESCRIPTION
**Pull request by szabolcs-instructure**
Share Extension Dashboard Notification logic. 

refs: MBL-16144
affects: Student
release note: none
test plan: See ticket

## Screenshots

<table>
<tr><th>Uploading a new submission</th></tr>
<tr>
<td><video src="https://user-images.githubusercontent.com/110813321/191504313-c9e5ca91-3a67-4554-b1ed-347db16ba6aa.mp4" maxHeight=500></td>
</tr>
</table>

## Checklist

- [ ] Follow-up e2e test ticket created or not needed
- [x] A11y checked
- [x] Tested on phone
- [x] Tested on tablet
- [ ] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Core/Core/Dashboard/ViewModel/FileUploadNotificationCardItemViewModel.swift
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Core/Core/Extensions/InstIconExtensions.swift:423:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    
    
</details>
<details><summary>:x: XCTest failed: CoreTests/SubmissionCompletedNotificationSenderTests/testSuccessNotificationBeingSent</summary>


    Assertion Failure at SubmissionCompletedNotificationsSenderTests.swift:68: Asynchronous wait failed: Exceeded timeout of 0.1 seconds, with unfulfilled expectations: "completion event fire".
</details>


[MBL-16144](https://instructure.atlassian.net/browse/MBL-16144)

#### Affected Apps: Student
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Review by balintbartok — DISMISSED** [review]
QA + 1

File: Core/Core/Store/Store.swift [hunk 1]
```diff
@@ -75,6 +75,7 @@ public class Store<U: UseCase>: NSObject, NSFetchedResultsControllerDelegate {
     public private(set) var pending: Bool = false
     public private(set) var requested: Bool = false
     public private(set) var error: Error?
+    private var externalStoreChangeListener: AnyCancellable?
```
Comment (vargaat) [inline Core/Core/Store/Store.swift]: This property isn't used in this file.

**Review by vargaat — APPROVED** [review]
QA+1

**Review by davidvarga93 — APPROVED** [review]
QA + 1

**Review by balintbartok — APPROVED** [review]
QA + 1

**Pull request merged**